homeassistant/components/sensor/sense.py

```diff
@@ -0,0 +1,136 @@
+"""
+Support for monitoring a Sense energy sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sense/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_EMAIL, CONF_PASSWORD)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['sense_energy==0.3.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ACTIVE_NAME = "Energy"
+DAILY_NAME = "Daily"
+WEEKLY_NAME = "Weekly"
+MONTHLY_NAME = "Monthly"
+YEARLY_NAME = "Yearly"
+
+PRODUCTION_NAME = "Production"
+CONSUMPTION_NAME = "Usage"
+
+ACTIVE_TYPE = 'active'
+DAILY_TYPE = 'DAY'
+WEEKLY_TYPE = 'WEEK'
+MONTHLY_TYPE = 'MONTH'
+YEARLY_TYPE = 'YEAR'
+
+ICON = 'mdi:flash'
+
+MIN_TIME_BETWEEN_DAILY_UPDATES = timedelta(seconds=300)
+MIN_TIME_BETWEEN_ACTIVE_UPDATES = timedelta(seconds=60)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Sense sensor."""
+    from sense_energy import Senseable
+
+    username = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+
+    data = Senseable(username, password)
+
+    @Throttle(MIN_TIME_BETWEEN_DAILY_UPDATES)
+    def update_trends():
+        """Update the daily power usage."""
+        data.update_trend_data()
+
+    @Throttle(MIN_TIME_BETWEEN_ACTIVE_UPDATES)
+    def update_active():
+        """Update the active power usage."""
+        data.get_realtime()
+
+    add_devices([
+        # Active power usage/production
+        Sense(data, ACTIVE_NAME, ACTIVE_TYPE, False, update_active),
+        Sense(data, ACTIVE_NAME, ACTIVE_TYPE, True, update_active),
+        # Daily Power
+        Sense(data, DAILY_NAME, DAILY_TYPE, False, update_trends),
+        Sense(data, DAILY_NAME, DAILY_TYPE, True, update_trends),
+        # Weekly Power
+        Sense(data, WEEKLY_NAME, WEEKLY_TYPE, False, update_trends),
+        Sense(data, WEEKLY_NAME, WEEKLY_TYPE, True, update_trends),
+        # Monthly Power
+        Sense(data, MONTHLY_NAME, MONTHLY_TYPE, False, update_trends),
+        Sense(data, MONTHLY_NAME, MONTHLY_TYPE, True, update_trends),
+        # Yearly Power
+        Sense(data, YEARLY_NAME, YEARLY_TYPE, False, update_trends),
+        Sense(data, YEARLY_NAME, YEARLY_TYPE, True, update_trends),
```

Comment:
Instead of this use `monitored_conditions` which would let the users decide which sensors they want to use.